packages/client/test/l1.e2e.spec.ts

```diff
@@ -0,0 +1,265 @@
+/*
+  This test script (e2e.spec.ts) aims to perform integrated testing of the project. It executes a series of actions,
+  including deploying the contracts (registry, offchain resolver, and universal resolver), creating the Client using Viem, 
+  and initializing the gateway locally. After deploying and configuring the contracts, the Client can access
+  off-chain information during the tests. It's important to note that this initial test script only sets up the
+  environment and stops at the gateway call. It still requires implementing the connection between the gateway and 
+  layer two, or the gateway and the database.
+*/
+
+// Importing abi and bytecode from contracts folder
+import {
+  abi as abiL1Resolver,
+  bytecode as bytecodeL1Resolver,
+} from '@blockful/contracts/out/L1Resolver.sol/L1Resolver.json'
+import {
+  abi as abiOffchainResolver,
+  bytecode as bytecodeOffchainResolver,
+} from '@blockful/contracts/out/OffchainResolver.sol/OffchainResolver.json'
+import {
+  abi as abiDummyNameWrapper,
+  bytecode as bytecodeDummyNameWrapper,
+} from '@blockful/contracts/out/DummyNameWrapper.sol/DummyNameWrapper.json'
+import {
+  abi as abiL1Verifier,
+  bytecode as bytecodeL1Verifier,
+} from '@blockful/contracts/out/L1Verifier.sol/L1Verifier.json'
+import {
+  abi as abiRegistry,
+  bytecode as bytecodeRegistry,
+} from '@blockful/contracts/out/ENSRegistry.sol/ENSRegistry.json'
+import {
+  abi as abiUniversalResolver,
+  bytecode as bytecodeUniversalResolver,
+} from '@blockful/contracts/out/UniversalResolver.sol/UniversalResolver.json'
+
+import { normalize, labelhash, namehash } from 'viem/ens'
+import { anvil } from 'viem/chains'
+import {
+  Address,
+  GetContractReturnType,
+  Hash,
+  WalletClient,
+  createTestClient,
+  getContract,
+  getContractAddress,
+  http,
+  publicActions,
+  walletActions,
+  zeroHash,
+} from 'viem'
+import { expect } from 'chai'
+
+import { L1ProofService } from '@blockful/gateway/src/services'
+import { NewApp } from '@blockful/gateway/src/app'
+import { withGetStorageSlot } from '@blockful/gateway/src/handlers'
+
+const GATEWAY_URLS = ['http://127.0.0.1:3000/{sender}/{data}.json']
+
+let l1Resolver: GetContractReturnType<
+  typeof abiL1Resolver,
+  { wallet: WalletClient }
+>
+let offchainResolverAddr: Hash, universalResolverAddr: Hash
+
+const client = createTestClient({
+  chain: anvil,
+  mode: 'anvil',
+  transport: http(),
+})
+  .extend(walletActions)
+  .extend(publicActions)
+
+async function deployContract({
+  abi,
+  bytecode,
+  account,
+  args,
+}: {
+  abi: unknown[]
```

Comment:
Instead of unknown, maybe we can use something like https://abitype.dev/api/types#abi.

Reply:
it doesn't match what is generated by `forge build`